### ISSUE TITLE
Abandoned project?

### ISSUE THREAD

**Issue by SPYFF** [amluto/virtme#86]
Hi @amluto ! 

The number of unmerged PRs and the last commit's date indicates that the project is no longer maintained. Is there any plan to continue the maintenance or invite someone else to maintain it and merge PRs?

Its quite unfortunate that this project is hosted on git.kernel.org which (for me at least) indicates its a safe bet for its purpose. However looking around in the PRs there are lots of useful features unmerged and fixes for breaking errors with newer kernels.

Am I missing something? Is there any successor of this project and got deprecated?

Thanks in advance!
Ferenc

**Comment by marcosps** [comment]
@SPYFF I've being trying to reach him for quite some time, asking if I could help to maintain virtme, but without success...

**Comment by SPYFF** [comment]
@marcosps That is unfortunate but thanks for your effort nevertheless. 
A looked around and found [kernelcraft](https://github.com/arighi/kernelcraft) by @arighi from Canonical with similar functionalities. It claims itself as a modified/extended version of virtme, however I have no idea about its feature parity since I just find that. Nor whether it will be official tool from Canonical or its a personal project which might abandoned at some time.

If you can somehow get access to the maintenance of virtme that would be the best, somehow drive similar efforts to the same direction. The other option might be help to @arighi to achieve feature parity with virtme (again, if thats not the case currently) and hopefully on the way he might give maintainer permission for more person if he decide to discontinue the project. And forking virtme always an option but every source on the web will continue to points to this seemingly abandoned project.

**Comment by marcosps** [comment]
I've being getting patches on my forked version, specially ones for s390 and ppc64 (since I used on those archs too):

https://github.com/marcosps/virtme

Maybe I should start getting more patches in my fork?

**Comment by SPYFF** [comment]
Well probably that would works too.

Also you can take a look into other efforts around the project, there might be some useful commit without PR: https://github.com/amluto/virtme/network

**Comment by arighi** [comment]
kernelcraft is not an official Canonical project (not yet at least), I started it as a personal project, collecting a bunch of custom scripts written on top of virtme that I was using to bisect kernel issues. Then it ended up in a little more complex project aiming at quickly recompiling & testing kernels. I decided to "fork" my own version virtme, because I also noticed that new PRs are stuck waiting for review and I needed to apply some of them (and also apply some custom changes to match kernelcraft requirements), so I could just move faster by forking the official virtme. But I'd be really happy to help integrate all the missing features in the official virtme, it's an amazing project and it'd be really sad if it's abandoned.

**Comment by SPYFF** [comment]
Thank you for the clarification and the effort you put into your virtme fork. Is it possible to somehow funnel the pull requests from here into your fork, or thats illegal/unethical? I totally understand if you dont want to bother with that too.

**Comment by arighi** [comment]
I'll be happy to look at the PRs and try to apply them to my fork, at the moment I was only taking those that I personally needed (hoping that at some point they were merged upstream into the official virtme). I don't think it's illegal/unethical to apply upstream PRs into a fork, but It'd be nice to get a feedback on this from @amluto ...

**Comment by marcosps** [comment]
> I'll be happy to look at the PRs and try to apply them to my fork, at the moment I was only taking those that I personally needed (hoping that at some point they were merged upstream into the official virtme). I don't think it's illegal/unethical to apply upstream PRs into a fork, but It'd be nice to get a feedback on this from @amluto ...

If you want, I can also help getting the patches. Avoiding double work here is my main focus, and since you're using virtme as base of kernelcraft, I can help getting the fixes targetting virtme, at least until @amluto appears again, or maybe your fork can be the nextr "virtme-ng" :)

**Comment by SPYFF** [comment]
Thanks folks for doing it, ! I plan to add VirtioFS and qemu-micro VM support  and hopefully on the road I can help to test/fix some bugs.

**Comment by marcosps** [comment]
@arighi would you kindly get other PRs that are around here for some time? Some related to s390 and ppc specially? (I would need those since I teste on those machine...) :D

**Comment by arighi** [comment]
@marcosps sure! I was planning to review all the PR here and merge those that make sense in my virtme fork. Between today and tomorrow I'll try to catch up with all of them.

**Comment by arighi** [comment]
Would it be ok if I leave a comment in the PRs here to keep track of what has been applied to my fork and what not? Any objection?

**Comment by arighi** [comment]
@marcosps BTW, I think I merged all your s390/ppc-related PRs (and tested on these arches). Can you double check if I'm missing anything important? Thanks!

**Comment by SPYFF** [comment]
> Would it be ok if I leave a comment in the PRs here to keep track of what has been applied to my fork and what not? Any objection?

Absolutely, hopefully others can find the active fork as well with that notice.

**Comment by marcosps** [comment]
> @marcosps BTW, I think I merged all your s390/ppc-related PRs (and tested on these arches). Can you double check if I'm missing anything important? Thanks!

s390 worked like expected, but for ppc64le I had some problems. First, you changed your fork to always default to ppc64le, but the qemu binary is always called ppc64. So changing it locally now gives me different errors, but that I would check later with some ppc on SUSE, since I had to add more ```-machine``` options to make it run, and even with the options it gets stuck, without showing any output at all...

I'll submit one fix for the qemu binary name in your fork. Thanks a lot for your work!
